### ISSUE TITLE
Show mods for the current major Minecraft version doesn't work on CurseForge

### ISSUE THREAD

**Issue by Banaando** [ATLauncher/ATLauncher#647]
**Describe the bug**
Duplicate of https://github.com/ATLauncher/ATLauncher/issues/629#issue-1291881294 but it only happens on CurseForge. Modrinth works fine. From my testing (see screenshots) I'm assuming it happens on every Minecraft version for every modloader.

**To Reproduce**
1. Enable Show mods for the current major Minecraft version (eg: 1.16.x) in the Settings tab.
2. Search for a mod that isn't available on the modloader or version your modpack is on.
3. The mod will show up anyways.

**Expected behavior**
Mods only available for the current major Minecraft version for your modloader show up.

**Logs**
Nothing out of the ordinary in the logs.

**Screenshots**
Quark showing up for 1.19.2 Fabric
![image](https://user-images.githubusercontent.com/67934979/186553824-befba1f6-a514-4293-947f-b56a02fe5ed7.png)
The Aether mod showing up on 1.19.2 Fabric
![image](https://user-images.githubusercontent.com/67934979/186553890-92572f07-3491-4681-aae2-a87a4aecd4a5.png)
The Aether mod showing up on 1.19.2 Forge
![image](https://user-images.githubusercontent.com/67934979/186554013-5183f9a2-9fd7-4f3d-8a59-4b25853aeeb1.png)
The Aether mod showing up on 1.14.4 Forge
![image](https://user-images.githubusercontent.com/67934979/186554146-a8d06b17-69b6-43e2-8013-be27e47f7646.png)
KronHUD showing up on 1.12.2 Forge
![image](https://user-images.githubusercontent.com/67934979/186554452-fbf22aac-2131-4303-9efb-165f96a4d2f3.png)

**System information (please complete the following information, or remove if not applicable):**
- OS: Windows 10
- Java Version: The one built into the launcher.

**Comment by RyanTheAllmighty** [comment]
Yeah their old api didn't support searching by major version, but the new one does and just hasn't been implemented yet
